DOPS-171: Update translation process with weblate changes. (#3780)
mattermost/docs: source/developer/localization.rst

2 - Confirm that the language you want to help translate is listed in the shipped project of the `Translation Server <https://translate.mattermost.com/projects/mattermost/mattermost-server_master/>`__.
If it is, perhaps offer to help translate or review and make suggestions.

3 - Confirm that the language you want to help translate is listed in the work in progress project of the `Translation Server <https://translate.mattermost.com/projects/i18n-wip/mattermost-server-wip/>`__.
If it is, perhaps offer to help translate or review and make suggestions.

4 - The language you want to help translate is neither in the shipped project nor in the work in progress projects `projects <https://translate.mattermost.com/projects/>`__.

5 - Review translation rules written by localization leads, when applicable:

    - `French (Règles pour la traduction francophone de Mattermost) <https://github.com/wget/mattermost-localization-french-translation-rules>`__
    - `German (Regeln zur Übersetzung von Mattermost) <https://gist.github.com/meilon/1317a9425988b3ab296c894a72270787>`__

Translations Updates
--------------------

> 1 - New and updated source strings are being automatically pulled on github pushes by developers.

Reviewer: ```suggestion
1 - New and updated source strings are being automatically pulled on GitHub pushes by developers.
```

> 2 - A new pull request with latest translations that reach at least Beta Quality is submitted to the `mattermost-mobile <https://github.com/mattermost/mattermost-mobile>`__, `mattermost-server <https://github.com/mattermost/mattermost-server>`__ and `mattermost-webapp <https://github.com/mattermost/mattermost-webapp>`__ repos each Monday at 22:00 GMT.

Reviewer: ```suggestion
2 - A new pull request with latest translations that reach at least Beta quality is submitted to the `mattermost-mobile <https://github.com/mattermost/mattermost-mobile>`__, `mattermost-server <https://github.com/mattermost/mattermost-server>`__ and `mattermost-webapp <https://github.com/mattermost/mattermost-webapp>`__ repos each Monday at 22:00 GMT.
```

> 1 - Join the `Mattermost Translation Server <http://translate.mattermost.com>`__.

Reviewer 2: Propose simplifying

```
1 - Join the `Mattermost Translation Server <http://translate.mattermost.com>`__.

2 - Confirm that the language you want to help translate is listed in the `Translation Server <https://translate.mattermost.com/projects/>`_. The language may be listed as shipped or work in progress, depending on if the language `ships in Mattermost <https://docs.mattermost.com/help/settings/account-settings.html#language>`_.

    - If it is, perhaps offer to help translate or review and make suggestions.
    - If not, ask to setup your language on the `Mattermost localization channel <https://community.mattermost.com/core/channels/localization>`__, so you can start helping with the translation.

3 - Review translation rules written by localization leads, when applicable:

    - `French (Règles pour la traduction francophone de Mattermost) <https://github.com/wget/mattermost-localization-french-translation-rules>`__
    - `German (Regeln zur Übersetzung von Mattermost) <https://gist.github.com/meilon/1317a9425988b3ab296c894a72270787>`__
```

2 - Confirm that the language you want to help translate is listed in the shipped project of the `Translation Server <https://translate.mattermost.com/projects/mattermost/mattermost-server_master/>`__.
If it is, perhaps offer to help translate or review and make suggestions.

3 - Confirm that the language you want to help translate is listed in the work in progress project of the `Translation Server <https://translate.mattermost.com/projects/i18n-wip/mattermost-server-wip/>`__.
If it is, perhaps offer to help translate or review and make suggestions.

> 4 - The language you want to help translate is neither in the shipped project nor in the work in progress projects `projects <https://translate.mattermost.com/projects/>`__.

Reviewer: ```suggestion
4 - The language you want to help translate is neither in the shipped project nor in the work in progress `projects <https://translate.mattermost.com/projects/>`__.
```

5 - Review translation rules written by localization leads, when applicable:

    - `French (Règles pour la traduction francophone de Mattermost) <https://github.com/wget/mattermost-localization-french-translation-rules>`__
    - `German (Regeln zur Übersetzung von Mattermost) <https://gist.github.com/meilon/1317a9425988b3ab296c894a72270787>`__

Translations Updates
--------------------

> 1 - New and updated source strings are being automatically pulled on GitHub pushes by developers.

Reviewer 2: What does `being automatically pulled on GitHub pushes by developers` mean?

Reviewer 3: Same. Added.

> 2 - A new pull request with latest translations that reach at least Beta quality is submitted to the `mattermost-mobile <https://github.com/mattermost/mattermost-mobile>`__, `mattermost-server <https://github.com/mattermost/mattermost-server>`__ and `mattermost-webapp <https://github.com/mattermost/mattermost-webapp>`__ repos each Monday at 22:00 GMT.

Reviewer 2: Do we still have PRs submitted once per week on Mondays?

Author: With some new changes I did, it should.

Reviewer 2: Awesome!

1 - Join the `Mattermost Translation Server <http://translate.mattermost.com>`__.

> 2 - Confirm that the language you want to help translate is listed in the `Translation Server <https://translate.mattermost.com/projects/>`_. The language may be listed as shipped or work in progress, depending on if the language `ships in Mattermost <https://docs.mattermost.com/help/settings/account-settings.html#language>`_.

Reviewer: ```suggestion
2 - Confirm that the language you want to help translate is listed in the `Translation Server <https://translate.mattermost.com/projects/>`_. The language may be listed as shipped or work in progress, depending whether the language `ships in Mattermost <https://docs.mattermost.com/help/settings/account-settings.html#language>`_.
```

Translations Updates
--------------------

> 1 - New and updated source strings are being automatically added to the translation server on GitHub pushes by developers.

Reviewer 2: @metanerd To confirm, do you mean new and updated source strings are automatically added to the translation service each time a relevant GitHub PR is merged?

Author: Yes, I do mean exactly that.

Author: Should I use your phrasing?

Reviewer 2: @metanerd yes, lets

> 1 - New and updated source strings are being automatically added to the translation server on GitHub pushes by developers.

Reviewer 2: ```suggestion
1 - New and updated source strings are automatically added to the translation server each time a GitHub PR containing source strings is merged.
```

Reviewer 3: Added

> 2 - A new pull request with latest translations that reach at least Beta quality is submitted to the `mattermost-mobile <https://github.com/mattermost/mattermost-mobile>`__, `mattermost-server <https://github.com/mattermost/mattermost-server>`__ and `mattermost-webapp <https://github.com/mattermost/mattermost-webapp>`__ repos each Monday at 22:00 GMT.

Reviewer: ```suggestion
2 - A new pull request with latest translations that reach at least Beta quality is submitted to the `mattermost-mobile <https://github.com/mattermost/mattermost-mobile>`__, `mattermost-server <https://github.com/mattermost/mattermost-server>`__, and `mattermost-webapp <https://github.com/mattermost/mattermost-webapp>`__ repos each Monday at 22:00 GMT.
```

1 - Build Mattermost on your machine following the `Developer Machine Setup Guides <https://docs.mattermost.com/developer/dev-setup.html>`__.

2 - Download a copy of your translations to your local machine by going to the language of your choice, e.g. `German <https://translate.mattermost.com/projects/mattermost/mattermost-server_master/de/>`_, then **Files > Download original translation file**.

3 - Copy the generated ``[locale].json`` files to the corresponding directories:

    - For the server, copy the files to the i18n directory of the ``mattermost-server`` project.
    - For the webapp, copy the files to the i18n directory of the ``mattermost-webapp`` project.

4 - Modify the file ``i18n/i18n.jsx`` in the ``mattermost-webapp`` project to include your translated strings.

5 - Compile and run **Mattermost** to confirm everything works. You can then review and verify translations from your machine.

> If you find a string that has not been translated, search for the string in the respective localization file to confirm it's included for translations. You can find the English version for server, webapp and mobile projects below

Reviewer: ```suggestion
If you find a string that has not been translated, search for the string in the respective localization file to confirm it's included for translations. You can find the English version for server, webapp, and mobile projects below
```

1 - Build Mattermost on your machine following the `Developer Machine Setup Guides <https://docs.mattermost.com/developer/dev-setup.html>`__.

> 2 - Download a copy of your translations to your local machine by going to the language of your choice, e.g. `German <https://translate.mattermost.com/projects/mattermost/mattermost-server_master/de/>` -> Files -> Download original translation file (JSON file)

Reviewer: ```suggestion
2 - Download a copy of your translations to your local machine by going to the language of your choice, e.g. `German <https://translate.mattermost.com/projects/mattermost/mattermost-server_master/de/>` > Files > Download original translation file (``.json`` file).
```

> 2 - Download a copy of your translations to your local machine by going to the language of your choice, e.g. `German <https://translate.mattermost.com/projects/mattermost/mattermost-server_master/de/>` > Files > Download original translation file (``.json`` file).

Reviewer 2: ```suggestion
2 - Download a copy of your translations to your local machine by going to the language of your choice, e.g. `German <https://translate.mattermost.com/projects/mattermost/mattermost-server_master/de/>`_, then **Files > Download original translation file**.
```

> 3 - Copy the generated [locale].json files to the corresponding directories:

Reviewer: ```suggestion
3 - Copy the generated ``[locale].json`` files to the corresponding directories:
```

    "Türkçe - Turkish", "`Kaya Zeren <https://twitter.com/kaya_zeren>`__", ""
    "Pусский - Russian", "`Alexey Napalkov <https://github.com/flynbit>`_", ""
    "Yкраїнська - Ukrainian", "TBD (Open Role)", ""

>     "한국어 - Korean", "`avasconcelos114 <https://translate.mattermost.com/user/avasconcelos114/>`__", ""

Reviewer 2: Confirming Andre agreed to be the official reviewer/translator for Korean?

Author: Actually no, you are right. Removed.

    "中文 (简体) - Simplified Chinese", "`aeomin <http://translate.mattermost.com/user/aeomin/>`__", ""
    "中文 (繁體) - Traditional Chinese", "`Tze-Kei Lee (chikei) <https://github.com/chikei>`__", ""
    "日本語 - Japanese", "`Yusuke Nemoto (kaakaa) <https://github.com/kaakaa>`__", ""

Administrative tasks
~~~~~~~~

> To grant trusted translators additional permissions as weblate admin, add the user to the mattermost@TrustedReviewers Group in weblate.

Reviewer: ```suggestion
To grant trusted translators additional permissions as Weblate admin, add the user to the *mattermost@TrustedReviewers* group in Weblate.
```

> Admin interface (tool icon in weblate) > Users > Django Admin Interface > click on the User to add to the group > Groups > Add to mattermost@TrustedReviewers > Save.

Reviewer: ```suggestion
**Admin interface (tool icon in Weblate) > Users > Django Admin Interface > select the User > Groups > Add to *mattermost@TrustedReviewers* > Save**.
```
